### PR TITLE
Update Kickstarter/Reactive-Extensions

### PR DESCRIPTION
**Pull request by msadoon**
### Pull Request Description

Updating commit hash for building Kickstarter/Reactive-Swift with Swift 5.0.

@swift-ci Please test source compatibility

### Acceptance Criteria

To be accepted into the Swift source compatibility test suite, a project must:

- [x] be an *Xcode* or *swift package manager* project
- [x] support building on either Linux or macOS
- [x] target Linux, macOS, or iOS/tvOS/watchOS device
- [x] be contained in a publicly accessible git repository
- [x] maintain a project branch that builds against Swift 4.0 and passes any unit tests
- [x] have maintainers who will commit to resolve issues in a timely manner
- [x] be compatible with the latest GM/Beta versions of *Xcode* and *swiftpm*
- [x] add value not already included in the suite
- [x] be licensed with one of the following permissive licenses:
	* BSD
	* MIT
	* Apache License, version 2.0
	* Eclipse Public License
	* Mozilla Public License (MPL) 1.1
	* MPL 2.0
	* CDDL
- [x] pass `./project_precommit_check` script run

Ensure project meets all listed requirements before submitting a pull request.

### CODE REVIEW

**Comment by justice-adams-apple** [comment]
@swift-ci test

**Comment by justice-adams-apple** [comment]
Hi @msadoon , thanks for the PR. A few things:
1.) You will likely want to update your [4.0 revision](https://github.com/apple/swift-source-compat-suite/pull/710/files#diff-a962f52a12fff6ca1f89224230f941bd5ac6c666aec8df5cd2402599a8bf5c4bR1194) as well. Feel free to use the same project revision `e3f7786`. This will allow you to fix https://github.com/apple/swift/issues/57433 and will ensure your project is tested properly with changes and remains compatible with swift 4.0

2.) After that, you will want to remove all the xfail blocks associated with your project since we no longer expect any build failures 😄 . Without removing the xfails, you will get a `UPASS` result which will fail the CI build (as what happened here)

* [Xfail 1](https://github.com/apple/swift-source-compat-suite/pull/710/files#diff-a962f52a12fff6ca1f89224230f941bd5ac6c666aec8df5cd2402599a8bf5c4bR1212)
* [Xfail 2](https://github.com/apple/swift-source-compat-suite/pull/710/files#diff-a962f52a12fff6ca1f89224230f941bd5ac6c666aec8df5cd2402599a8bf5c4bR1227)

**Comment by msadoon** [comment]
Hi @justice-adams-apple ,
I made the changes you suggested, but my precommit check for version 4.0 is failing locally with the same hash as for 5.0.
Do I need to rebuild the project in Xcode just for Swift 4.0 and make a commit? The previous commit hash passed one test but not the second one. 
Just overall curious about how I might start to patch this issue.

File: projects.json [hunk 1]
```diff
@@ -1191,11 +1191,11 @@
     "compatibility": [
       {
         "version": "4.0",
```
Comment (justice-adams-apple) [inline projects.json:1193]: @msadoon
>I made the changes you suggested, but my precommit check for version 4.0 is failing locally with the same hash as for 5.0.
Do I need to rebuild the project in Xcode just for Swift 4.0 and make a commit? 

Looks like the new hash for your project is minimum Swift 4.2 compatible. Go ahead and bump the string here from `4.0` to `4.2` and I suspect your build will work

Comment (msadoon) [reply projects.json]: Done! Works, thank you!

**Comment by justice-adams-apple** [comment]
@swift-ci test

**Review by justice-adams-apple — APPROVED** [review]
LGTM. Thank you @msadoon !

**Pull request merged**